Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -113,14 +114,32 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithm::execute(
       slAccessorDelegate;
   slAccessorDelegate.connect<&IndexSourceLinkAccessor::range>(&slAccessor);
 
-  Acts::PropagatorPlainOptions pOptions;
-  pOptions.maxSteps = m_cfg.maxSteps;
-  pOptions.direction = Acts::Direction::Forward;
+  Acts::PropagatorPlainOptions firstPropOptions;
+  firstPropOptions.maxSteps = m_cfg.maxSteps;
+  firstPropOptions.direction = Acts::Direction::Forward;
+
+  Acts::PropagatorPlainOptions secondPropOptions;
+  secondPropOptions.maxSteps = m_cfg.maxSteps;
+  secondPropOptions.direction = firstPropOptions.direction.invert();
 
   // Set the CombinatorialKalmanFilter options
-  ActsExamples::TrackFindingAlgorithm::TrackFinderOptions options(
+  ActsExamples::TrackFindingAlgorithm::TrackFinderOptions firstOptions(
       ctx.geoContext, ctx.magFieldContext, ctx.calibContext, slAccessorDelegate,
-      extensions, pOptions);
+      extensions, firstPropOptions);
+
+  // For the second pass we do not want to branch which is the default behavior
+  // of the MeasurementSelector
+  Acts::MeasurementSelector secondMeasSel{Acts::MeasurementSelector::Config{
+      {Acts::GeometryIdentifier(),
+       Acts::MeasurementSelectorCuts{{}, {5}, {1}}}}};
```

Comment:
```suggestion
  Acts::MeasurementSelector::Config secondMeasSelCfg{m_cfg.measurementSelectorCfg};
  for (auto cuts : m_cfg.measurementSelectorCfg) {
    cuts.numMeasurementsCutOff = {1};
  }
  Acts::MeasurementSelector secondMeasSel{std::move(secondMeasSelCfg)};
```
or
```suggestion
  std::vector<Acts::MeasurementSelector::Config::InputElement> secondMeasSelCfg;
  secondMeasSelCfg.resize(m_cfg.measurementSelectorCfg.size());
  for (std::size_t index = 0; index < m_cfg.measurementSelectorCfg.size(); ++index) {
    secondMeasSelCfg.emplace_back(
        m_cfg.measurementSelectorCfg.idAt(index),
        Acts::MeasurementSelectorCuts{
            m_cfg.measurementSelectorCfg.valueAt(index).etaBins,
            m_cfg.measurementSelectorCfg.valueAt(index).chi2CutOff,
            {1}});
  }
  Acts::MeasurementSelector secondMeasSel{std::move(secondMeasSelCfg)};
```
to save a `vector` allocation by copying each member. (Note, it is OK to specify a single value, even if we have eta bins.)

Or did you mean to have a different `chi2CutOff` as well? The default is `15` (in `reconstruction.py`), but the user could change it.

Reply:
Wow! It looks like you removed the need for this suggestion at just the same time I posted it 😁

Reply:
Oh yeah sorry ... I just realized that this is still in there. That was a workaround to only process a single branch in the second pass. But with the recent changes it should be possible again to branch there too.